### PR TITLE
Refactor the code for Compatibility with FeynmanDiagram.jl v1.0

### PR DESCRIPTION
**Pull request by houpc**
TODO:
1. benchmark vertex4 results
2. update free energy, vertex3, and green
3. improve code testing

### CODE REVIEW

**Review by dcerkoney — APPROVED** [review]

File: src/freeEnergy/freeEnergy.jl [hunk 1]
```diff
@@ -1,95 +1,166 @@
-module FreeEnergy
```
Comment (dcerkoney) [inline src/freeEnergy/freeEnergy.jl:1]: This file still needs to be cleaned up with `dos2unix`.

File: src/vertex4/vertex4.jl [hunk 2]
```diff
@@ -163,14 +116,74 @@ end
...
+# include("ver4_PH_l_vegas.jl")
+# include("ver4_PH_l_mcmc.jl")
+# include("ver4_ParquetAD_compile_dynamic.jl")
+# include("ver4KW.jl")
+
+include("source_codeParquetAD/Cwrapper_vertex4_ParquetAD.jl")
+include("source_codeParquetAD_Proper/Cwrapper_vertex4_ParquetAD.jl")
+
+const evalfunc_vertex4_map = Dict(
+    (0, 0, 0) => eval_vertex4_ParquetAD000!,
```
Comment (dcerkoney) [inline src/vertex4/vertex4.jl:134]: I notice this pattern creates a lot of code duplication.
Why not define a function like the following in module `Vertex4`:
```julia
function evalfunc_vertex4_map(partition::Tuple{Int,Int,Int})
    getfield(Main, Symbol("eval_vertex4_ParquetAD", join(P), "!"))
end
```
with analogous definitions for other observables, and then do `evalfunc_vertex4_map(P)(root, leafVal)`?

**Review by dcerkoney — APPROVED** [review]
Looks good to me, modulo a few comments

File: example/sigmadk_compile.jl [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+_partition = UEG.partition(order)
+
+partition = Vector{NTuple{4,Int}}()
+for p in _partition
+    push!(partition, (p..., 1))
+end
+FeynGraphs = Diagram.diagram_parquet_noresponse(:sigma, para, partition, [pr -> pr isa FrontEnds.BareGreenId,
+        pr -> pr isa FrontEnds.BareInteractionId, pr -> pr.extK[1] != 0], optimize_level=1)
+
+Diagram.compileC_ParquetAD_toFiles(FeynGraphs, order + 1, "sigmadk", compiler="icc")
```
[the remark this answers is not in the document]
Comment (houpc) [reply example/sigmadk_compile.jl:16]: The choice of compiler should depend on the user's computer. In general, the running speed compiled by icc is faster than that of gcc.

File: src/vertex4/vertex4.jl (hunk 2 above)
Comment (houpc) [reply src/vertex4/vertex4.jl]: I'm not sure which way is better. I think the constant dictionary mapping may be more predictable and easier for Julia's JIT compiler to optimize with direct loopup and no reflection. The mappings here are relatively static and the set of functions is manageable, so the code duplication is relatively acceptable.

Comment (dcerkoney) [reply src/vertex4/vertex4.jl]: I agree. Benchmarking with a toy function showed that your current implementation is an order of magnitude faster, so let's stick to it. I agree, it is an acceptable amount.

**Pull request merged**